bigchaindb/backend/mongodb/connection.py

```diff
@@ -41,9 +46,90 @@ def db(self):
     def _connect(self):
         for i in range(self.max_tries):
             try:
-                self.connection = MongoClient(self.host, self.port)
-            except ConnectionFailure as exc:
+                # we should only return a connection if the replica set is
+                # initialized. initialize_replica_set will check if the
+                # replica set is initialized else it will initialize it.
+                initialize_replica_set()
+                self.connection = MongoClient(self.host, self.port,
+                                              replicaset=self.replicaset)
+            except errors.ConnectionFailure as exc:
                 if i + 1 == self.max_tries:
                     raise
                 else:
                     time.sleep(2**i)
+
+
+def initialize_replica_set():
+    """Initialize a replica set. If already initialized skip."""
+
+    # Setup a MongoDB connection
+    # The reason we do this instead of `backend.connect` is that
+    # `backend.connect` will connect you to a replica set but this fails if
+    # you try to connect to a replica set that is not yet initialized
+    conn = MongoClient(host=bigchaindb.config['database']['host'],
+                       port=bigchaindb.config['database']['port'])
+    _check_replica_set(conn)
+    config = {'_id': bigchaindb.config['database']['replicaset'],
+              'members': [{'_id': 0, 'host': 'localhost:27017'}]}
```

Comment:
I wonder: should we get the host and port from the config, for cases where the host and port differ from defaults?